lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -144,7 +144,96 @@ TODO
 
 ## Using exceptions for control-flow
 
-TODO
+#### Problem
+
+This anti-pattern refers to code that forces developers to handle exceptions for control flow. Exception handling itself does not represent an anti-pattern, but this should not be the only alternative available to developers to handle an error in client code. When developers have no freedom to decide if an error is exceptional or not, this is considered an anti-pattern.
+
+#### Example
+
+An example of this anti-pattern, as shown below, is when a library forces its users to use `try/1` statements to rescue raised exceptions and handle different cases. Such a library library does not allow developers to decide if an error is exceptional or not in their applications.
+
+```elixir
+defmodule MyModule do
+  def janky_function(value) do
+    if is_integer(value) do
+      "It's an integer"
+    else
+      raise "expected integer, got: #{inspect(value)}"
+    end
```

Comment:
I suggest finding another example, where the error is not for the developer about types. This particular case can be refactored with the guard, which provides a great error message:
```elixir
def janky_function(value) when is_integer(value) do
  "It's an integer"
end
```
so someone may think that using guards is antipattern, and it is better to use ok/error pairs.

Also, I think, it is worth highlighting that even if the function has ok/error signature, it is okay to raise exceptions if the error is about invalid usage of the function API by the developer.

Reply:
One suggestion is to write a function that performs a fake HTTP call (such as `make_http_call()` and raises if the response status is not 200. That's a great use case for non-bang + bang variants. Thoughts?

Reply:
I think fake HTTP call is a great alternative.